### PR TITLE
adding more errors & modification on the code to be more swifty code.

### PR DESCRIPTION
**Pull request by SalehAlDhobaie**
the modification in the code was : 
- Change `check` function to be static in the struct.
- Change the calculation of sum to `reduce` to be more swifty code.
- Rearrange the validation conditions.
- Adding a new Error Type for validation.

### CODE REVIEW

**Comment by alhazmy13** [comment]
Thanks @SalehAlDhobaie For your effort
 Can you please remove all print lines to accept your PR?

File: validateSAID_SWIFT-3_.swift [hunk 1]
```diff
@@ -1,28 +1,85 @@
+//: Playground - noun: a place where people can play
+
+import UIKit
```
Comment (alhazmy13) [inline validateSAID_SWIFT-3_.swift]: Do you really need this import?

File: validateSAID_SWIFT-3_.swift [hunk 2]
```diff
@@ -1,28 +1,85 @@
...
+                //print("\n\n \(#function), \(#line) value: \(idString.charAt(at: i))")
+                let ZFOdd = String(format: "%02d", Int(String(idString.charAt(at: i)))! * 2)
+                sum1 += Int(String(ZFOdd.charAt(at: 0)))! + Int(String(ZFOdd.charAt(at: 1)))!
+            } else {
+                sum1 += Int(String(idString.charAt(at: i)))!
+            }
+        }
+        
+        print("\n\n \(#function), \(#line) value: \(sum1)")
+        print("\n\n \(#function), \(#line) value: \(reduced)")
```
Comment (alhazmy13) [inline validateSAID_SWIFT-3_.swift]: Thanks @SalehAlDhobaie For your effort
Can you please remove all print lines to accept your PR?

**Comment by SalehAlDhobaie** [comment]
Thanks !

**Pull request merged**